### PR TITLE
WIP: Implement DropDownButton

### PR DESCRIPTION
**Pull request by WinXaito**
See #85 

image

image

image

## Pre-launch Checklist

- [x] I have run `dartfmt` on all changed files  
- [x] I have updated `CHANGELOG.md` with my changes  
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could
- [x] I have added/updated relevant documentation
- [x] I have run `flutter pub publish --dry-run` and addressed any warnings

*Note: currently in WIP, I don't have added any documentation or commentary, I wait to see if the widget is accepeted*

### CODE REVIEW

File: lib/src/controls/inputs/dropdown_button.dart [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  final FocusNode? focusNode;
+
+  /// {@macro flutter.widgets.Focus.autofocus}
+  final bool autofocus;
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasFluentTheme(context));
+
+    final buttonChilds = <Widget>[];
```
Comment (NicolaVerbeeck) [inline lib/src/controls/inputs/dropdown_button.dart:45]: buttonChildren

File: lib/src/controls/inputs/dropdown_button.dart [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    );
+  }
+}
+
+class DropDownButtonItem {
+  DropDownButtonItem({
+    this.leading,
+    this.title,
+    this.trailing,
+    required this.onTap,
```
Comment (NicolaVerbeeck) [inline lib/src/controls/inputs/dropdown_button.dart:104]: put required arguments first

File: lib/src/controls/inputs/dropdown_button.dart [hunk 3]
```diff
@@ -0,0 +1,111 @@
+import 'package:fluent_ui/fluent_ui.dart';
+
+const double _kDefaultPadding = 5.0;
+const double _kVerticalOffset = 20.0;
+const double _kContentWidth = 100.0;
+
+class DropDownButton extends StatelessWidget {
+  const DropDownButton({
+    Key? key,
+    required this.items,
```
Comment (NicolaVerbeeck) [inline lib/src/controls/inputs/dropdown_button.dart:10]: Put required arguments first (and place Key last)

Comment (WinXaito) [reply lib/src/controls/inputs/dropdown_button.dart:10]: I checked on other files, Key is everytime in first position.

File: lib/src/controls/inputs/dropdown_button.dart [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    required this.onTap,
+    this.leading,
+    this.title,
+    this.trailing,
+  }) : assert(leading != null || title != null || trailing != null);
+
+  final Widget? leading;
+  final Widget? title;
+  final Widget? trailing;
+  final Function() onTap;
```
Comment (bdlukaa) [inline lib/src/controls/inputs/dropdown_button.dart:110]:
```suggestion
  final VoidCallback onTap;
```

Comment (WinXaito) [reply lib/src/controls/inputs/dropdown_button.dart]: Fixed

**Review by NicolaVerbeeck — APPROVED** [review]

File: lib/src/controls/inputs/dropdown_button.dart [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        child: leading,
+      ));
+    }
+    if (title != null) {
+      buttonChildren.add(title!);
+    }
+    buttonChildren.add(Padding(
+      padding: EdgeInsets.only(left: padding?.left ?? _kDefaultPadding),
+      child: trailing ?? const Icon(FluentIcons.chevron_down, size: 12),
+    ));
```
Comment (bdlukaa) [inline lib/src/controls/inputs/dropdown_button.dart:58]: For better readability:
```suggestion
    final buttonChildren = <Widget>[
      if (leading != null)
        Padding(
          padding: EdgeInsets.only(right: padding?.right ?? _kDefaultPadding),
          child: leading,
        ),
      if (title != null) title!,
      Padding(
        padding: EdgeInsets.only(left: padding?.left ?? _kDefaultPadding),
        child: trailing ?? const Icon(FluentIcons.chevron_down, size: 12),
      ),
    ];
```

File: lib/src/controls/inputs/dropdown_button.dart [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        onPressed: disabled ? null : () => controller.open = true,
+        autofocus: autofocus,
+        focusNode: focusNode,
+      ),
+    );
+  }
+}
+
+class DropDownButtonItem {
+  DropDownButtonItem({
```
Comment (bdlukaa) [inline lib/src/controls/inputs/dropdown_button.dart:100]:
```suggestion
  const DropDownButtonItem({
```

File: lib/src/controls/inputs/dropdown_button.dart [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    required this.onTap,
+    this.leading,
+    this.title,
+    this.trailing,
+  }) : assert(leading != null || title != null || trailing != null);
```
Comment (bdlukaa) [inline lib/src/controls/inputs/dropdown_button.dart:105]: Text could be better
```suggestion
  }) : assert(
    leading != null || title != null || trailing != null,
    'You must provide at least one property: leading, title or trailing'
  );
```

**Review by bdlukaa — CHANGES_REQUESTED** [review]
Nice implementation

File: lib/src/controls/inputs/dropdown_button.dart [hunk 8]
```diff
@@ -21,15 +26,42 @@ class DropDownButton extends StatelessWidget {
     this.autofocus = false,
   }) : super(key: key);
```
Comment (bdlukaa) [inline lib/src/controls/inputs/dropdown_button.dart:27]:
```suggestion
  }) : assert(items.isNotEmpty, 'You must provide at least one item'),
       super(key: key);
```

**Review by bdlukaa — CHANGES_REQUESTED** [review]
Documentation needs to be updated.

`padding`, `verticalOffset` and `contentWidth` values could be passed on the constructor, making it possible to make them non-nullable

**Review by bdlukaa — APPROVED** [review]

**Pull request merged**